music21/converter/subConverters.py

```diff
@@ -364,11 +367,30 @@ def show(self, obj, fmt, app=None, subformats=None, **keywords):  # pragma: no c
                 fp = helperSubConverter.write(s, helperFormat,
                                               subformats=helperSubformats, **keywords)
 
-                if helperSubformats[0] == 'png':
+                if helperSubformats[0] in ('png', 'svg'):
                     if not str(environLocal['musescoreDirectPNGPath']).startswith('/skip'):
                         ipo = ipythonObjects.IPythonPNGObject(fp)
                         # noinspection PyTypeChecker
-                        display(Image(data=ipo.getData(), retina=True))
+                        if helperSubformats[0] == 'svg':
+                            svgData = ipo.getData().decode('utf8')
+                            # musescore appears to scale svg as 3.6 x png scale
+                            svgScale = keywords.get("svgScale", float(environLocal["ipythonSVGScale"]))
+                            if svgScale:
+                                def _scale(mo):
+                                    begin, w, h, end = mo.groups()
+                                    w, h = float(w)*svgScale, float(h)*svgScale
+                                    return f'{begin}width="{w}px" height="{h}px"{end}'
+                                svgData = re.sub(
+                                    r'^(<svg[^>]*)width\s*=\s*"([\d.]+)px"\s+height\s*=\s*"([\d.]+)px"([^>]*>)',
+                                    _scale, svgData, flags=re.MULTILINE)
```

Comment:
All this is too fragile -- it depends on width preceding height, etc.  -- it should use ElementTree or another built in xml parser to extract and change the values.  Please don't define a function inside an if statement.

Reply:
I admit it does look fragile but that said the musescore C++ always generates the attributes in that order. See

https://github.com/musescore/MuseScore/blob/9e456031cf1e843aab2a96322d6c3efbd92816a2/mscore/svggenerator.cpp#L1036

That said, I can pick those attributes out one-by-one although I won't use a full blown xml parser as this would slow things down and potentially cause headaches with namespaces etc. Let me know if you will be happy with that?

I'm happy to move the function out of the "if" scope although that will mean it will become a partial function.

Reply:
@mscuthbert just awaiting your reply to my comment

Reply:
It doesn't matter what MuseScore does today -- if they change in the future or there are successor tools, etc., I'm the one who is going to need to read this regexp and figure out how to fix it 8 years from now.  I've learned from experience, I only put into music21 code I can read and I can fix, because eventually I'll have to.  :-)

Reply:
Then I'm sorry about this. I don't really have time to work an xml parser into this. I'll leave the PR here for now for someone else to pick up if interested.

Best wishes on your project.